dot/parachain/gossip-support/gossip_support.go

```diff
@@ -71,19 +119,133 @@ func (*GossipSupport) Name() parachaintypes.SubSystemName {
 	return parachaintypes.GossipSupport
 }
 
-func (*GossipSupport) ProcessActiveLeavesUpdateSignal(signal parachaintypes.ActiveLeavesUpdateSignal) error {
-	// TODO implement in #4507
+func (gs *GossipSupport) ProcessActiveLeavesUpdateSignal(signal parachaintypes.ActiveLeavesUpdateSignal) error {
+	logger.Trace("Process ActiveLeavesUpdateSignal")
+
+	leaf := signal.Activated.Hash
+
+	rt, err := gs.blockState.GetRuntime(leaf)
+	if err != nil {
+		return err
+	}
+
+	currentIndex, err := rt.ParachainHostSessionIndexForChild()
+	if err != nil {
+		return err
+	}
+
+	sinceFailure := time.Duration(0)
+	if gs.lastFailure != nil {
+		sinceFailure = time.Since(*gs.lastFailure)
+	}
+
+	sinceLastReconnect := time.Duration(0)
+	if gs.lastConnectionRequest != nil {
+		sinceLastReconnect = time.Since(*gs.lastConnectionRequest)
+	}
+
+	forceRequest := sinceFailure >= BackoffDuration
+	reResolveAuthorities := sinceLastReconnect >= TryReResolveAuthorities
+	ls := &leafSession{
+		currentIndex,
+		leaf,
+	}
+
+	maybeNewSession := ls
+	if gs.lastSessionIndex != nil && currentIndex <= *gs.lastSessionIndex {
+		maybeNewSession = nil
+	}
+
+	maybeIssueConnection := maybeNewSession
+	if forceRequest || reResolveAuthorities {
+		maybeIssueConnection = ls
+	}
+
+	if maybeIssueConnection != nil {
+		sessionIndex := maybeIssueConnection.currentIndex
+		relayParent := maybeIssueConnection.Leaf
+
+		sessionInfo, err := rt.ParachainHostSessionInfo(sessionIndex)
+		if err != nil {
+			logger.Warnf("failed to get session info for session %d", sessionIndex)
+			return err
+		}
```

Comment:
Here, they check if `session_info` is None

```suggestion
		sessionInfo, err := rt.ParachainHostSessionInfo(sessionIndex)
		if err != nil ||  sessionInfo == nil {
			logger.Warnf("failed to get session info for session %d", sessionIndex)
			return err
		}
```